src/comm.cc

```diff
@@ -131,6 +131,7 @@ comm_udp_recvfrom(int fd, void *buf, size_t len, int flags, Ip::Address &from)
     int x = recvfrom(fd, buf, len, flags, AI->ai_addr, &AI->ai_addrlen);
     from = *AI;
     Ip::Address::FreeAddr(AI);
+    fd_table[fd].bytesRead(x);
```

Comment:
I think we should not add this line into this low-level function because this low-level function cannot be certain that:

* comm_udp_recvfrom() is (and will always be) called for an fd_table-registered FD
* callers do not (and will never) call bytesRead() themselves
 
Instead, let the callers of low-level functions continue to update the byte counters until all such calls disappear behind fde or we find a better/safer way to address this uncertainty. Yes, we will miss a few calls, but that is a lesser evil (for now).

Please check for other similar cases in the PR -- I have not.

Reply:
One of the guarantees with FDs is that they are unique for the duration of their lifetime. If the FD is not registered the corresponding fde will be reserved and showing `closed` state. Recording traffic for them regardless of state is a good idea to avoid some of the bugs being fixed here recurring and harmless in the case the fde object was not actively `open`ed.

Reply:
> quite a lot of new calls to bytesRead/bytesWritten with XXX markers are added as a result of this request

Addressing the problem identified in this change request did _not_ require the addition of those calls.

I am not against improving UDP accounting, including improving UDP accounting by adding those (temporary) calls with XXXs, but that improvement is a _separate_ issue/decision. I have not asked for that accounting improvement and have not even suggested it. In fact, I have suggested the _opposite_ approach -- "keeping a few (translated) legacy calls" that had to be kept to avoid accounting regressions, resulting in "virtually no new bytesRead() calls".